climada/hazard/base.py

```diff
@@ -369,6 +371,361 @@ def set_vector(self, *args, **kwargs):
+            return x.values
+
+        def strict_positive_int_accessor(x: xr.DataArray, report_key: str):
+            """Only allow positive, non-zero integers"""
+            values = x.values
+            if not np.issubdtype(values.dtype, np.integer):
+                raise TypeError(f"'{report_key}' data array must be integers")
+            if not np.all(values > 0):
+                raise ValueError(f"'{report_key}' data must be larger than zero")
+            return np.asarray(values, dtype=np.int64)
+
+        # Create a DataFrame storing access information for each of data_vars
+        num_events = data.sizes["event"]
+        keys = ["fraction", "frequency", "event_id", "event_name", "date"]
+        data_ident = pd.DataFrame(
+            data=dict(
+                # The attribute of the Hazard class where data is stored
+                hazard_attr=keys,
+                # The identifier and default key used in this method
+                identifier=keys,
+                # The keys assigned by the user
+                user_key=None,
+                # The default value for each attribute
+                default_value=[
+                    to_csr_matrix(
+                        xr.apply_ufunc(
+                            lambda x: np.where(x != 0, 1, 0), data[intensity]
+                        ).values
+                    ),
+                    np.ones(num_events),
+                    np.array(range(num_events), dtype=int) + 1,
+                    list(data[coords["time"]].values),
+                    np.array(u_dt.datetime64_to_ordinal(data[coords["time"]].values)),
+                ],
+                # The accessor for the data in the Dataset
+                accessor=[
+                    lambda x: to_csr_matrix(default_accessor(x)),
+                    default_accessor,
+                    lambda x: strict_positive_int_accessor(x, "event_id"),
+                    lambda x: list(default_accessor(x)),  # list, not np.array
+                    lambda x: strict_positive_int_accessor(x, "date"),
+                ],
+            )
+        )
+
+        # Update the keys from user settings
+        if data_vars is not None:
+            ident = data_ident["identifier"]
+            unknown_keys = [
+                key for key in data_vars.keys() if not ident.str.contains(key).any()
+            ]
+            if unknown_keys:
+                raise ValueError(
+                    f"Unknown data variables passed: '{unknown_keys}'. Supported "
+                    f"data variables are {list(ident)}."
+                )
+
+            # Make 'identifier' an index for replacement, then make it a column again
+            data_ident = data_ident.set_index("identifier")
+            data_ident["user_key"].loc[list(data_vars.keys())] = list(
+                data_vars.values()
+            )
+            data_ident.reset_index(inplace=True)
+
+        def read_or_default(ident: pd.Series):
```

Comment:
I do not understand the naming of this method. I also do not understand the docstring.

Reply:
I updated it to `load_data_or_default`, improved the signature and wrote a proper docstring, see https://github.com/CLIMADA-project/climada_python/pull/507/commits/9c68a9e67b60e71e10c811759de0ad506c0eaa6a